diff -ruN clean/kernel/supercalls.c work/kernel/supercalls.c
--- a/kernel/supercalls.c	2026-03-02 18:36:28.895068295 +0100
+++ b/kernel/supercalls.c	2026-03-03 19:26:43.515530216 +0100
@@ -345,7 +345,9 @@
     ret = ksu_set_app_profile(&cmd.profile);
     if (!ret) {
         ksu_persistent_allow_list();
+#ifndef CONFIG_KSU_SUSFS
         ksu_mark_running_process();
+#endif
     }
     return ret;
 }
@@ -492,12 +494,12 @@
 
     ret = strncpy_from_user(mnt, cmd.arg, sizeof(mnt));
     if (ret < 0) {
-        pr_err("nuke ext4 copy mnt failed: %ld\\n", ret);
-        return -EFAULT; // 或者 return ret;
+        pr_err("nuke ext4 copy mnt failed: %ld\n", ret);
+        return -EFAULT;
     }
 
     if (ret == sizeof(mnt)) {
-        pr_err("nuke ext4 mnt path too long\\n");
+        pr_err("nuke ext4 mnt path too long\n");
         return -ENAMETOOLONG;
     }
 

diff -ruN clean/kernel/Kconfig work/kernel/Kconfig
--- a/kernel/Kconfig	2026-03-04 01:26:28.645687237 +0100
+++ b/kernel/Kconfig	2026-03-04 01:26:28.646572683 +0100
@@ -33,4 +33,92 @@
     select KALLSYMS
     select KALLSYMS_ALL
 
+menu "KernelSU - SUSFS"
+config KSU_SUSFS
+    bool "KernelSU addon - SUSFS"
+    depends on KSU
+    depends on THREAD_INFO_IN_TASK
+    default y
+    help
+        Patch and Enable SUSFS to kernel with KernelSU.
+
+config KSU_SUSFS_SUS_PATH
+    bool "Enable to hide suspicious path"
+    depends on KSU_SUSFS
+    default y
+    help
+        Allow hiding the user-defined path and all its sub-paths from various system calls.
+
+config KSU_SUSFS_SUS_MOUNT
+    bool "Enable to hide suspicious mounts"
+    depends on KSU_SUSFS
+    default y
+    help
+        Allow hiding the user-defined mount paths from /proc/self/[mounts|mountinfo|mountstat].
+
+config KSU_SUSFS_SUS_KSTAT
+    bool "Enable to spoof suspicious kstat"
+    depends on KSU_SUSFS
+    default y
+    help
+        Allow spoofing the kstat of user-defined file/directory.
+
+config KSU_SUSFS_SUS_KSTAT_REDIRECT
+    bool "Enable to redirect suspicious kstat"
+    depends on KSU_SUSFS_SUS_KSTAT
+    default y
+    help
+        Allow redirecting kstat checks to a different target path.
+
+config KSU_SUSFS_SUS_MAP
+    bool "Enable to hide suspicious memory maps"
+    depends on KSU_SUSFS
+    default y
+    help
+        Allow hiding mmapped real file from /proc/pid/maps and related interfaces.
+
+config KSU_SUSFS_SPOOF_UNAME
+    bool "Enable to spoof uname"
+    depends on KSU_SUSFS
+    default y
+    help
+        Allow spoofing the string returned by uname syscall.
+
+config KSU_SUSFS_ENABLE_LOG
+    bool "Enable logging susfs log to kernel"
+    depends on KSU_SUSFS
+    default y
+    help
+        Allow logging susfs log to kernel.
+
+config KSU_SUSFS_SPOOF_CMDLINE_OR_BOOTCONFIG
+    bool "Enable to spoof /proc/bootconfig or /proc/cmdline"
+    depends on KSU_SUSFS
+    default y
+    help
+        Spoof the output of /proc/bootconfig (gki) or /proc/cmdline (non-gki).
+
+config KSU_SUSFS_OPEN_REDIRECT
+    bool "Enable to redirect open path"
+    depends on KSU_SUSFS
+    default y
+    help
+        Allow redirecting a target path to be opened with another user-defined path.
+
+config KSU_SUSFS_HIDE_KSU_SUSFS_SYMBOLS
+    bool "Enable to hide ksu and susfs symbols from /proc/kallsyms"
+    depends on KSU_SUSFS
+    default y
+    help
+        Automatically hide ksu and susfs symbols from /proc/kallsyms.
+
+config KSU_SUSFS_UNICODE_FILTER
+    bool "Enable unicode character filter"
+    depends on KSU_SUSFS
+    default y
+    help
+        Enable filtering of unicode characters in paths.
+
+endmenu
+
 endmenu
